src/main/cpp/util/md5.cc

```diff
@@ -209,10 +210,11 @@ void Md5Digest::Transform(
   // FF, GG, HH, and II transformations for rounds 1, 2, 3, and 4.
   // Rotation is separate from addition to prevent recomputation.
 #define FF(a, b, c, d, s, ac) { \
-      (a) += F((b), (c), (d)) + ((*x_pos++ = *cur_word++)) + \
+      (a) += F((b), (c), (d)) + ((*x_pos++ = htole32(*cur_word))) + \
```

Comment:
This one interprets the input bytes as the same 32-bit number regardless of architecture, right? (technically, it should be le32toh, but the reverse transformation is the same as the forward one)

Reply:
Yes, `le32toh()` is better here, since it reads the input bytes in little-endian format and stores them in host byte order.

Reply:
Please add a comment that explains that `htole32()` and `le32toh()` are actually the same thing... oh wait, you can't do that, this is a preprocessor macro. Oh well.

Reply:
Added above.